iothub/device/src/Transport/Mqtt/SimpleWorkQueue.cs

```diff
@@ -111,9 +111,8 @@ public virtual void Abort(Exception exception)
                 case States.FinalProcessing:
                     State = States.Aborted;
 
-                    while (_backlogQueue.Any())
+                    while (_backlogQueue.Any() && _backlogQueue.TryDequeue(out TWork workItem))
```

Comment:
I'm wondering if `_backlogQueue.Any()` is needed here.

Reply:
You are correct, not needed.

Reply:
There is one a little further down in the PR too.